projects/igniteui-angular/src/lib/directives/for-of/for_of.sync.service.ts

```diff
@@ -45,4 +45,11 @@ export class IgxForOfSyncService {
     public chunkSize(dir: string): number {
         return this._master.get(dir).state.chunkSize;
     }
+
+    /**
+     * @hidden
+     */
+    public scrollPosition(dir: string): number {
+        return this._master.get(dir).scrollPosition;
```

Comment:
@ChronosSF 
Storing the scroll position in the master directive may be a problem for scenarios where the rows are cached, since it’s possible for the master directive that caches the position to be detached while the scrolling occurs, hence it will not detect any changes until it is re-added in the view.
As a result data cells may become de-synched from the scrollbar hence the headers and content will be misaligned.

Here’s an example scenario where can occurs:
1.	Run the gridGroupBy sample.
For the rows the first one is the one holding the “master” igxForOf that stores the position.
2.	Group By a column.
First row becomes group row, hence the data row is now cached (is in detached state) hence the “master” igxForOf directive is also detached.
3.	Scroll right.
4.	Ungroup
The data row is back from the cache, so is the igxForOf “master”. However it has retained its old scroll position and has forced all other rows to match it.

Result is that the header/footer cells and the data row cells are misaligned now, since data row cells are not synched with the scrollbar.